probe-rs/src/debug/mod.rs

```diff
@@ -236,8 +245,22 @@ impl<'debuginfo, 'probe, 'core> Iterator for StackFrameIterator<'debuginfo, 'pro
 
             use gimli::read::RegisterRule::*;
 
-            self.registers[i] = match unwind_info.register(gimli::Register(i as u16)) {
-                Undefined => None,
+            let register_rule = unwind_info.register(gimli::Register(i as u16));
+
+            log::trace!("Register {}: {:?}", i, &register_rule);
+
+            self.registers[i] = match register_rule {
+                Undefined => {
+                    // If we get undefined for the LR register (register 14) or any callee saved register,
+                    // we assume that it is unchanged. Gimli doesn't allow us
+                    // to distinguish if  a rule is not present or actually set to Undefined
+                    // in the call frame information.
+
+                    match i {
+                        4 | 5 | 6 | 7 | 8 | 10 | 11 | 14 => self.registers[i],
```

Comment:
How does one know what those numbers mean? Are those defined for DWARF?

Reply:
DWARF uses numbered registers. Each architecture then has to define for itself which number maps to which register.

The whole register implementation is not really generic right now in the debug implementation, it's very ARM specific.